lib/finch/http1/pool.ex

```diff
@@ -75,21 +75,20 @@ defmodule Finch.HTTP1.Pool do
     {:ok, {conn, idle_time}, conn}
   end
 
-  def handle_checkout(:checkout, {pid, _}, conn) do
+  def handle_checkout(:checkout, _from, conn) do
```

Comment:
For me, this is still behaving the same as the last commit.
```suggestion
  def handle_checkout(:checkout, _from, conn) do
    IO.puts("CHECKOUT")
```
Add this line to see NimblePool cycle through each worker on the pool size + 1 request. I used your exact example to reproduce: 
```
size = 10
Finch.start_link(name: F1, pools: %{default: [size: size, count: 1]})
request = Finch.build(:get, "http://localhost:80/wait/10")

for i <- 1..(size+1) do
  IO.puts i
  IO.inspect(Finch.request(request, F1))
  receive do
    msg -> IO.inspect msg
  after
    0 -> :ok
  end
end
```
After running this, wait a minute or two, then `flush()` and you will see the `{:tcp_closed, #Port<0.1079>}` message.

Reply:
I think this is expected. The connection will be closed eventually if it is not being used.

Reply:
But the message is being sent to the client, and unhandled, so the Mint struct is not updated so we don't know that it has been closed until it fails when we try to use it again. If we transferred the connection to the pool, then handle_info would be called, and the worker would be removed.

Reply:
Oh, I missed that the message is being leaked. If you are using this PR no socket messages should be sent (excluding bugs) when the connection is checked out since we set the socket to passive mode before.

Reply:
Additionally, the socket is never transferred so only the pool process should receive that message.

Reply:
I am using this PR, but we are still setting the socket to active mode when we check the connection back in to the pool, and since the controlling process is always the client, since this PR remove the transfer code, the client will be the one receiving the messages while in active mode, right?

Reply:
I don't believe so, when removing the transfer code the pool should always be the controlling process since the connection is started in the pool.

Reply:
Actually, we open the connection in the client: https://github.com/keathley/finch/blob/59978c535baff72953b221eb7ea8c3548b8b5375/lib/finch/http1/pool.ex#L42

Reply:
Ah shit. I forgot about this. If we open it in the client, then we definitely need to transfer it the first time around. My bad, I will look into this now.